lib/rbac/filterer.rb

```diff
@@ -281,9 +285,13 @@ def safe_base_class(klass)
 
     def rbac_class(scope)
       klass = scope.respond_to?(:klass) ? scope.klass : scope
-      return klass if apply_rbac_to_class?(klass)
-      if apply_rbac_to_associated_class?(klass)
-        return klass.name[0..-12].constantize.base_class # e.g. VmPerformance => VmOrTemplate
+      return klass if apply_rbac_directly?(klass)
+      if apply_rbac_through_association?(klass)
+        # Strip "Performance" off class name, which is the associated model
+        # of that metric.
+        # e.g. HostPerformance => Host
+        #      VmPerformance   => VmOrTemplate
+        return klass.name[0..-12].constantize.base_class
```

Comment:
Can we just ask the class what class to base rbac on?  This whole `directly?` vs. `through association/formerly indirectly?`  seems like it could be simpler by just asking the class these questions.